farftp/src/Ftp.cpp

```diff
@@ -271,10 +271,12 @@ int FTP::GetFindData(PluginPanelItem **pPanelItem, int *pItemsNumber, int OpMode
 		{
 			if(!NewPanelItem)
 				NewPanelItem = (PluginPanelItem *)malloc((1024+1)*sizeof(PluginPanelItem));
-			else
-				NewPanelItem = (PluginPanelItem *)realloc(NewPanelItem,(*pItemsNumber+1024+1)*sizeof(PluginPanelItem));
-
-			if(NewPanelItem == NULL)
+                        else {
+                                PluginPanelItem *tmp  = (PluginPanelItem *)realloc(NewPanelItem,(*pItemsNumber+1024+1)*sizeof(PluginPanelItem));
+                                if ( tmp ) NewPanelItem = tmp;
+                                else {free(NewPanelItem); NewPanelItem == nullptr;}
```

Comment:
Comparing instead of assignment: ==
Also put free and NewPanelItem=nulptr on separate lines